### PR TITLE
feat: mediator  added more logs and associated message Hash and structured logging

### PR DESCRIPTION
**Pull request by mineme0110**
ATL-4932

### CODE REVIEW

File: mediator/src/main/scala/io/iohk/atala/mediator/actions/ProtocolExecute.scala [hunk 1]
```diff
@@ -147,7 +147,7 @@ object BasicMessageExecuter extends ProtocolExecuter[Any] {
   override def program[R1 <: Any](plaintextMessage: PlaintextMessage) = for {
     job <- BasicMessage.fromPlaintextMessage(plaintextMessage) match
       case Left(error) => ZIO.fail(MediatorDidError(FailToParse(error)))
-      case Right(bm)   => Console.printLine(bm.toString).mapError(ex => MediatorThrowable(ex))
+      case Right(bm)   => Console.printLine(bm.toString).mapError(ex => MediatorThrowable(ex))
```
Comment (FabioPinheiro) [inline mediator/src/main/scala/io/iohk/atala/mediator/actions/ProtocolExecute.scala]: Can you undo this?

File: mediator/src/main/scala/io/iohk/atala/mediator/app/MediatorAgent.scala [hunk 2]
```diff
@@ -114,36 +114,39 @@ case class MediatorAgent(
...
-                      case Some(from) => didSocketManager.update { _.link(from.asFROMTO, socketID) }
-                // TODO Store context of the decrypt unwarping
-                // TODO SreceiveMessagetore context with MsgID and PIURI
-                protocolHandler <- ZIO.service[ProtocolExecuter[Services]]
-                ret <- protocolHandler
-                  .execute(plaintextMessage)
-                  .tapError(ex => ZIO.logError(s"Error when execute Protocol: $ex"))
-              } yield ret
-        } yield maybeSyncReplyMsg
+        ZIO
```
Comment (FabioPinheiro) [inline mediator/src/main/scala/io/iohk/atala/mediator/app/MediatorAgent.scala:117]: this indentation also seems weird

File: mediator/src/main/scala/io/iohk/atala/mediator/app/MediatorAgent.scala [hunk 3]
```diff
@@ -275,5 +280,5 @@ object MediatorAgent {
           } else h
         )
       )
-    )
+    )
```
Comment (FabioPinheiro) [inline mediator/src/main/scala/io/iohk/atala/mediator/app/MediatorAgent.scala:284]: Undo this

**Review by FabioPinheiro — APPROVED** [review]
I approve but there are some things that need to be changed.
I'm also afraid scalafmt is not running on the CI

File: mediator/src/main/scala/io/iohk/atala/mediator/actions/ProtocolExecute.scala (hunk 1 above)
Comment (shaileshp0110) [reply mediator/src/main/scala/io/iohk/atala/mediator/actions/ProtocolExecute.scala]: yes sure

**Pull request merged**